lib/iex/lib/iex/evaluator.ex

```diff
@@ -10,33 +10,51 @@ defmodule IEx.Evaluator do
     * keeping expression history
 
   """
-  def start(server, leader) do
+  def start(server, leader, opts) do
     old_leader = Process.group_leader
     Process.group_leader(self, leader)
 
     try do
-      loop(server, IEx.History.init)
+      loop(server, IEx.History.init, run_state(opts))
     after
       Process.group_leader(self, old_leader)
     end
   end
 
-  defp loop(server, history) do
+  defp loop(server, history, state) do
     receive do
-      {:eval, ^server, code, state} ->
-        {result, history} = eval(code, state, history)
+      {:eval, ^server, code, iex_state} ->
+        {result, history, state} = eval(code, iex_state, history, state)
         send server, {:evaled, self, result}
-        loop(server, history)
+        loop(server, history, state)
+      {:peek_env, receiver} ->
+        send receiver, {:peek_env, state.env}
+        loop(server, history, state)
       {:done, ^server} ->
         :ok
     end
   end
 
-  @doc """
-  Locates and loads an .iex.exs file from one of predefined locations.
-  Returns the new state.
-  """
-  def load_dot_iex(state, path \\ nil) do
+  defp run_state(opts) do
```

Comment:
Sorry, minor nitpick: we should call it `loop_state` now or `evaluator_state`. :) It was called `run_state` before because it was the state of the `run` function (confusing name, sorry).